httpcore5/src/main/java/org/apache/hc/core5/http/nio/support/classic/SharedOutputBuffer.java

```diff
@@ -162,4 +164,11 @@ private void waitFlush() throws InterruptedIOException {
         setInputMode();
     }
 
+    private void propagateEndStreamOnce() throws IOException {
```

Comment:
@malaysf One really minor thing. I suppose just `propagateEndStream` as the method name should be clear enough.

Reply:
Sure, I'll make the change shortly